cmd/agent/main.go

```diff
@@ -42,33 +43,72 @@ const (
 )
 
 func main() {
-	if _, err := reconcileKubelet(context.Background()); err != nil {
-		logrus.Warnf("failed to reconcile kubelet, error: %v", err)
-	}
+	var err error
 
-	logrus.SetOutput(colorable.NewColorableStdout())
-	logserver.StartServerWithDefaults()
-	if os.Getenv("CATTLE_DEBUG") == "true" || os.Getenv("RANCHER_DEBUG") == "true" {
-		logrus.SetLevel(logrus.DebugLevel)
-	}
+	if len(os.Args) > 1 {
+		err = runArgs()
+	} else {
+		if _, err := reconcileKubelet(context.Background()); err != nil {
+			logrus.Warnf("failed to reconcile kubelet, error: %v", err)
+		}
 
-	initFeatures()
+		logrus.SetOutput(colorable.NewColorableStdout())
+		logserver.StartServerWithDefaults()
+		if os.Getenv("CATTLE_DEBUG") == "true" || os.Getenv("RANCHER_DEBUG") == "true" {
```

Comment:
Instead of having a ton of `os.Getenv` calls, could we just parse all of this into an `AgentConfig` struct that parses all of the environment variables before running an actual command that gets passed into the commands itself?

This would avoid having to do a ton of stuff around helper functions defined below since we can just reference the value from methods defined on `AgentConfig` itself, e.g. `getHostPathFromContaienr`, `getHostPathFromHost`, and `getContainerPath`.